static/hosts.js

```diff
@@ -0,0 +1,111 @@
+(function () {
+	// Storage for variables in page context.
+	$.hosts = {};
+
+	function addrSort(a, b) {
+		if (a.IPAddress < b.IPAddress) return -1;
+		if (a.IPAddress > b.IPAddress) return 1;
+		return 0;
+	}
+
+	function storageSort(a, b) {
+		return a.TotalStorage - b.TotalStorage;
+	}
+
+	function priceSort(a, b) {
+		return a.Price - b.Price;
+	}
+
+	// Converts a count of bytes into a formatted string describing that number
+	// in standard disk storage units (default is GB).
+	function formatStorage(storageBytes, format) {
+		var suffixes = ["B", "KB", "MB", "GB", "TB"]
+		if (suffixes.indexOf(format.toUpperCase()) < 0) {
+			format = "GB";
+		}
+		var suffixIndex = suffixes.indexOf(format.toUpperCase())
+		if (suffixIndex < 0) {
+			suffixIndex = 2;
+
+		}
+		var bytesPer = Math.pow(1000, suffixIndex);
+		return (storageBytes / bytesPer).toFixed(2) + " " + format;
+	}
+
+	// Deterimines the proper sorting function based on the URI query parameter
+	// (default is priceSort).
+	function getSortFunc() {
+		var query = URI.parseQuery(window.location.search);
+		var defaultSort = priceSort;
+		if (!("sort" in query)) {
+			return defaultSort;
+		}
+		switch (query["sort"]) {
+			case "addr":
+				return addrSort;
+			case "storage":
+				return storageSort;
+			case "price":
+			default:
+				return priceSort;
+		}
+	}
+
+	// Updates the page's URL without reloading the page.
+	function setLocation(uri) {
+		window.history.pushState({path:uri},'',uri);
+	}
+
+	// Sets the sorting mechanism for the table and updates the table with the
+	// specified sorting.
+	function setTableSorting(sortingValue) {
+		var uri = URI(window.location);
+		uri.setQuery("sort", sortingValue);
+		setLocation(uri.toString());
+		populateHostTable();
+	}
+
+	// Populates the table of known hosts.
+	function populateHostTable() {
+		$('#hostsTable > tbody').empty();
+		var hosts = $.hosts.hosts;
+		sortFunc = getSortFunc();
+		hosts.sort(sortFunc);
+		hosts.forEach(function (host) {
+			var storageFormatted = formatStorage(host.TotalStorage, "GB");
+			var priceFormatted = (host.Price / 1000000).toFixed(2) + " SC";
```

Comment:
Whoops, I thought it was 1 million H = 1 SC. I was [way off](http://stream1.gifsoup.com/view2/20140916/5109205/samsonite-i-was-way-off-o.gif).

Fixed!